snapcraft/internal/pluginhandler.py

```diff
@@ -656,15 +656,24 @@ def _migrate_files(snap_files, snap_dirs, srcdir, dstdir, missing_ok=False,
     for directory in snap_dirs:
         src = os.path.join(srcdir, directory)
         dst = os.path.join(dstdir, directory)
+
+        dir_stat = os.stat(src)
+        uid = dir_stat.st_uid
+        gid = dir_stat.st_gid
         os.makedirs(dst, exist_ok=True)
+        os.chown(dst, uid, gid, follow_symlinks=follow_symlinks)
         shutil.copystat(src, dst, follow_symlinks=follow_symlinks)
 
     for snap_file in snap_files:
         src = os.path.join(srcdir, snap_file)
         dst = os.path.join(dstdir, snap_file)
-        os.makedirs(os.path.dirname(dst), exist_ok=True)
-        shutil.copystat(os.path.dirname(src), os.path.dirname(dst),
-                        follow_symlinks=follow_symlinks)
+
+        dir_stat = os.stat(srcdir)
+        uid = dir_stat.st_uid
+        gid = dir_stat.st_gid
+        os.makedirs(dstdir, exist_ok=True)
+        shutil.copystat(srcdir, dstdir, follow_symlinks=follow_symlinks)
+        os.chown(dstdir, uid, gid, follow_symlinks=follow_symlinks)
```

Comment:
src and dst are files which we're not copying here.  Though if snap_file
has directory path components I'd need to use os.path.dirname(src) I guess.

On Thu, Aug 11, 2016 at 1:59 PM, Kyle Fazzari <notifications@github.com>
wrote:

> In snapcraft/internal/pluginhandler.py
> <https://github.com/snapcore/snapcraft/pull/723#discussion_r74471271>:
>
> >          shutil.copystat(src, dst, follow_symlinks=follow_symlinks)
> >
> >      for snap_file in snap_files:
> >          src = os.path.join(srcdir, snap_file)
> >          dst = os.path.join(dstdir, snap_file)
> > -        os.makedirs(os.path.dirname(dst), exist_ok=True)
> > -        shutil.copystat(os.path.dirname(src), os.path.dirname(dst),
> > -                        follow_symlinks=follow_symlinks)
> > +
> > +        dir_stat = os.stat(srcdir)
> > +        uid = dir_stat.st_uid
> > +        gid = dir_stat.st_gid
> > +        os.makedirs(dstdir, exist_ok=True)
> > +        shutil.copystat(srcdir, dstdir, follow_symlinks=follow_symlinks)
> > +        os.chown(dstdir, uid, gid, follow_symlinks=follow_symlinks)
>
> This section is operating upon srcdir and dstdir, when perhaps it should
> be operating upon src and dst.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/snapcore/snapcraft/pull/723/files/91fe951c6cdf7238af75f8df480ca1fdbfe6de49#r74471271>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnbyA5IiC3Z9bti9IwmufffdL_Y6gjfks5qe2MdgaJpZM4JiYV0>
> .
>